### PR TITLE
Added MolarMass, MolarVolume, InverseToMolarMass, and some extra Densities

### PR DESCRIPTION
**Pull request by Dokotela**
Added new properties
- MOLAR_MASS
- INVERSE_OF_MOLAR_MASS
- MOLAR_VOLUME

I also added some more densities. 

I've added tests for all of the above, but some of the conversion tests for MOLAR_VOLUME don't quite work the way I want them to. For instance:
MOLAR_VOLUME.nanoMolesPerMilliLiter: 999999.9999999999
But should be:
MOLAR_VOLUME.nanoMolesPerMilliLiter: 1000000

Any recommendations on where I went wrong?

### CODE REVIEW

**Comment by ferraridamiano** [comment]
Hi @Dokotela and thank you for another wonderful PR! I'm going to review it and commit if needed! Don't worry too much about that calculations problem. Hope to finish the review today

File: lib/properties/density.dart [hunk 1]
```diff
@@ -8,8 +8,23 @@ import 'package:units_converter/utils/utils.dart';
...
+  milliGramsPerMilliLiter(MASS.milligrams, VOLUME.milliliters),
+  milliGramsPerCubicMeter(MASS.milligrams, VOLUME.cubicMeters),
+  milliGramsPerCubicCentimeter(MASS.milligrams, VOLUME.cubicCentimeters),
+  microGramsPerLiter(MASS.micrograms, VOLUME.liters),
+  microGramsPerDeciLiter(MASS.micrograms, VOLUME.deciliters),
+  microGramsPerMilliLiter(MASS.micrograms, VOLUME.milliliters),
+  nanoGramsPerLiter(MASS.nanograms, VOLUME.liters),
+  nanoGramsPerMilliLiter(MASS.nanograms, VOLUME.milliliters),
+  picoGramsPerLiter(MASS.picograms, VOLUME.liters),
+  picoGramsPerMilliLiter(MASS.picograms, VOLUME.milliliters),
```
Comment (ferraridamiano) [inline lib/properties/density.dart:26]: Make these names more coherent. If a unit is written in muliple parts write them with camelCase
e.g. "milligrams per liter" -> milligramsPerLiter

File: lib/properties/density.dart [hunk 2]
```diff
@@ -48,6 +77,21 @@ class Density extends RatioProperty<DENSITY, MASS, VOLUME> {
...
+  Unit get milliGramsPerCubicMeter => getUnit(DENSITY.milliGramsPerCubicMeter);
+  Unit get milliGramsPerCubicCentimeter =>
+      getUnit(DENSITY.milliGramsPerCubicCentimeter);
+  Unit get microGramsPerLiter => getUnit(DENSITY.microGramsPerLiter);
+  Unit get microGramsPerDeciLiter => getUnit(DENSITY.microGramsPerDeciLiter);
+  Unit get microGramsPerMilliLiter => getUnit(DENSITY.microGramsPerMilliLiter);
+  Unit get nanoGramsPerLiter => getUnit(DENSITY.nanoGramsPerLiter);
+  Unit get nanoGramsPerMilliLiter => getUnit(DENSITY.nanoGramsPerMilliLiter);
+  Unit get picoGramsPerLiter => getUnit(DENSITY.picoGramsPerLiter);
+  Unit get picoGramsPerMilliLiter => getUnit(DENSITY.picoGramsPerMilliLiter);
```
Comment (ferraridamiano) [inline lib/properties/density.dart:96]: The same naming convention should also be applied here

File: lib/properties/molar_mass.dart [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  gramsPerPicoMole(MASS.grams, AMOUNT_OF_SUBSTANCE.picomoles),
+  gramsPerFemtoMole(MASS.grams, AMOUNT_OF_SUBSTANCE.femtomoles),
+  milliGramsPerMole(MASS.milligrams, AMOUNT_OF_SUBSTANCE.moles),
+  milliGramsPerMilliMole(MASS.milligrams, AMOUNT_OF_SUBSTANCE.millimoles),
+  milliGramsPerMicroMole(MASS.milligrams, AMOUNT_OF_SUBSTANCE.micromoles),
+  milliGramsPerNanoMole(MASS.milligrams, AMOUNT_OF_SUBSTANCE.nanomoles),
+  milliGramsPerPicoMole(MASS.milligrams, AMOUNT_OF_SUBSTANCE.picomoles),
+  milliGramsPerFemtoMole(MASS.milligrams, AMOUNT_OF_SUBSTANCE.femtomoles),
+  kiloGramsPerMole(MASS.kilograms, AMOUNT_OF_SUBSTANCE.moles),
+  kiloGramsPerMilliMole(MASS.kilograms, AMOUNT_OF_SUBSTANCE.millimoles),
```
Comment (ferraridamiano) [inline lib/properties/molar_mass.dart:21]: Also here

File: lib/properties/molar_mass.dart [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  Unit get gramsPerPicoMole => getUnit(MOLAR_MASS.gramsPerPicoMole);
+  Unit get gramsPerFemtoMole => getUnit(MOLAR_MASS.gramsPerFemtoMole);
+  Unit get milliGramsPerMole => getUnit(MOLAR_MASS.milliGramsPerMole);
+  Unit get milliGramsPerMilliMole => getUnit(MOLAR_MASS.milliGramsPerMilliMole);
+  Unit get milliGramsPerMicroMole => getUnit(MOLAR_MASS.milliGramsPerMicroMole);
+  Unit get milliGramsPerNanoMole => getUnit(MOLAR_MASS.milliGramsPerNanoMole);
+  Unit get milliGramsPerPicoMole => getUnit(MOLAR_MASS.milliGramsPerPicoMole);
+  Unit get milliGramsPerFemtoMole => getUnit(MOLAR_MASS.milliGramsPerFemtoMole);
+  Unit get kiloGramsPerMole => getUnit(MOLAR_MASS.kiloGramsPerMole);
+  Unit get kiloGramsPerMilliMole => getUnit(MOLAR_MASS.kiloGramsPerMilliMole);
```
Comment (ferraridamiano) [inline lib/properties/molar_mass.dart:82]: And here

File: lib/properties/molar_volume.dart [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  microMolesPerLiter(AMOUNT_OF_SUBSTANCE.micromoles, VOLUME.liters),
+  microMolesPerDeciLiter(AMOUNT_OF_SUBSTANCE.micromoles, VOLUME.deciliters),
+  microMolesPerMilliLiter(AMOUNT_OF_SUBSTANCE.micromoles, VOLUME.milliliters),
+  nanoMolesPerLiter(AMOUNT_OF_SUBSTANCE.nanomoles, VOLUME.liters),
+  nanoMolesPerDeciLiter(AMOUNT_OF_SUBSTANCE.nanomoles, VOLUME.deciliters),
+  nanoMolesPerMilliLiter(AMOUNT_OF_SUBSTANCE.nanomoles, VOLUME.milliliters),
+  picoMolesPerLiter(AMOUNT_OF_SUBSTANCE.picomoles, VOLUME.liters),
+  picoMolesPerDeciLiter(AMOUNT_OF_SUBSTANCE.picomoles, VOLUME.deciliters),
+  picoMolesPerMilliLiter(AMOUNT_OF_SUBSTANCE.picomoles, VOLUME.milliliters),
+  femtoMolesPerMilliLiter(AMOUNT_OF_SUBSTANCE.femtomoles, VOLUME.milliliters),
```
Comment (ferraridamiano) [inline lib/properties/molar_volume.dart:22]: Here

File: lib/properties/molar_volume.dart [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  Unit get nanoMolesPerLiter => getUnit(MOLAR_VOLUME.nanoMolesPerLiter);
+  Unit get nanoMolesPerDeciLiter => getUnit(MOLAR_VOLUME.nanoMolesPerDeciLiter);
+  Unit get nanoMolesPerMilliLiter =>
+      getUnit(MOLAR_VOLUME.nanoMolesPerMilliLiter);
+  Unit get picoMolesPerLiter => getUnit(MOLAR_VOLUME.picoMolesPerLiter);
+  Unit get picoMolesPerDeciLiter => getUnit(MOLAR_VOLUME.picoMolesPerDeciLiter);
+  Unit get picoMolesPerMilliLiter =>
+      getUnit(MOLAR_VOLUME.picoMolesPerMilliLiter);
+  Unit get femtoMolesPerMilliLiter =>
+      getUnit(MOLAR_VOLUME.femtoMolesPerMilliLiter);
```
Comment (ferraridamiano) [inline lib/properties/molar_volume.dart:92]: Here

File: lib/properties/molar_volume.dart [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    MOLAR_VOLUME.picoMolesPerMilliLiter: 'pmol/ml',
+    MOLAR_VOLUME.femtoMolesPerMilliLiter: 'fmol/ml',
+  };
+
+  ///Class for molar volume conversions, e.g. if you want to convert 1 gram per liter
+  ///in kilograms per liter:
+  ///```dart
+  ///var molarVolume = MolarVolume(removeTrailingZeros: false);
+  ///molarVolume.convert(Unit(MOLAR_VOLUME.gramsPerLiter, value: 1));
+  ///print(MOLAR_VOLUME.kilogramsPerLiter);
```
Comment (ferraridamiano) [inline lib/properties/molar_volume.dart:56]: The example refers to density

File: lib/properties/molar_mass.dart [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    MOLAR_MASS.kiloGramsPerMole: 'kg/mol',
+    MOLAR_MASS.kiloGramsPerMilliMole: 'kg/mmol',
+  };
+
+  ///Class for molar mass conversions, e.g. if you want to convert 1 gram per liter
+  ///in kilograms per liter:
+  ///```dart
+  ///var molarMass = MolarMass(removeTrailingZeros: false);
+  ///molarMass.convert(Unit(MOLAR_MASS.gramsPerLiter, value: 1));
+  ///print(MOLAR_MASS.kilogramsPerLiter);
```
Comment (ferraridamiano) [inline lib/properties/molar_mass.dart:53]: The example refers to density

File: lib/properties/reciprocal_of_molar_mass.dart [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    RECIPROCAL_OF_MOLAR_MASS.molesPerKiloGram: 'mol/kg',
+    RECIPROCAL_OF_MOLAR_MASS.milliMolesPerKiloGram: 'mmol/kg',
+  };
+
+  ///Class for the reciprocal of molar mass conversions, e.g. if you want to
+  ///convert 1 gram per liter into kilograms per liter:
+  ///```dart
+  ///var reciprocalOfMolarMass = MolarMass(removeTrailingZeros: false);
+  ///reciprocalOfMolarMass.convert(Unit(RECIPROCAL_OF_MOLAR_MASS.gramsPerLiter, value: 1));
+  ///print(RECIPROCAL_OF_MOLAR_MASS.kilogramsPerLiter);
```
Comment (ferraridamiano) [inline lib/properties/reciprocal_of_molar_mass.dart:54]: Update the example

File: lib/properties/reciprocal_of_molar_mass.dart [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+    RECIPROCAL_OF_MOLAR_MASS.femtoMolesPerGram: 'fmol/g',
+    RECIPROCAL_OF_MOLAR_MASS.picoMolesPerGram: 'pmol/g',
+    RECIPROCAL_OF_MOLAR_MASS.molesPerMilliGram: 'mol/mg',
+    RECIPROCAL_OF_MOLAR_MASS.milliMolesPerMilliGram: 'mmol/mg',
+    RECIPROCAL_OF_MOLAR_MASS.microMolesPerMilliGram: 'µmol/mg',
+    RECIPROCAL_OF_MOLAR_MASS.nanoMolesPerMilliGram: 'nmol/mg',
+    RECIPROCAL_OF_MOLAR_MASS.picoMolesPerMilliGram: 'pmol/mg',
+    RECIPROCAL_OF_MOLAR_MASS.femtoMolesPerMilliGram: 'fmol/mg',
+    RECIPROCAL_OF_MOLAR_MASS.molesPerKiloGram: 'mol/kg',
+    RECIPROCAL_OF_MOLAR_MASS.milliMolesPerKiloGram: 'mmol/kg',
```
Comment (ferraridamiano) [inline lib/properties/reciprocal_of_molar_mass.dart:46]: Also here

File: lib/properties/reciprocal_of_molar_mass.dart [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+  Unit get nanoMolesPerMilliGram =>
+      getUnit(RECIPROCAL_OF_MOLAR_MASS.nanoMolesPerMilliGram);
+  Unit get picoMolesPerMilligram =>
+      getUnit(RECIPROCAL_OF_MOLAR_MASS.picoMolesPerMilliGram);
+  Unit get femtoMolesPerMilliGram =>
+      getUnit(RECIPROCAL_OF_MOLAR_MASS.femtoMolesPerMilliGram);
+  Unit get molesPerKiloGram =>
+      getUnit(RECIPROCAL_OF_MOLAR_MASS.molesPerKiloGram);
+  Unit get milliMolesPerKiloGram =>
+      getUnit(RECIPROCAL_OF_MOLAR_MASS.milliMolesPerKiloGram);
```
Comment (ferraridamiano) [inline lib/properties/reciprocal_of_molar_mass.dart:96]: And here

**Review by ferraridamiano — CHANGES_REQUESTED** [review]
I submitted a cuple of commit in order to make the CI happy. The PR looks good. Most of the units name should be changed to be coherent with the other unit names (explained in a comment).

Now just a curiosity, is "reciprocal of molar mass" actually used in chemistry? Or is it something specific of your application?

Thank you again!

**Comment by Dokotela** [comment]
Above recommendations should be fixed. 

Reciprocal of molar mass isn't used in normal chemistry (at least I haven't seen it), but is used in medicine. 

If a patient needs 12 mmol of something, and the medication comes in 3 mmol/gram, you need to order 4 grams. Obviously a simple example.

**Comment by ferraridamiano** [comment]
Perfect, thank you very much @Dokotela! I will soon publish a new release with all your new units!

**Pull request merged**